Tests/L0/SonarQubePostTest/PRCA/PostCommentsTests.ps1

```diff
@@ -0,0 +1,244 @@
+        Assert-AreEqual $inputComment.RelativePath $thread.Properties[$itemPathName] "Invalid ItemPath property"
+        Assert-AreEqual $inputComment.Line $thread.Properties[$endLineName] "The EndLine should be the same as the comment's line"
+    }
+    
+    return $true
+}
+
+#
+# Validates static thread properties that are not tied to the original comment  
+#
+function ValidateCommonThreadAttributes
+{
+    param ([Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]$threadCollection)
+    
+    $startColumnName = [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadPropertyNames]::StartColumn
+    $positionContextName = [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadPropertyNames]::PositionContext
+    
+    $threadCollection | ForEach-Object {Assert-AreEqual "artifact uri" $_.ArtifactUri "Each thread should have the same ArtifactUri"}
+    $threadCollection | ForEach-Object {
+        Assert-AreEqual 
+        [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionStatus]::Active 
+        $_.Status 
+        "Each thread should be active"}
+    $threadCollection | ForEach-Object {Assert-AreEqual 1 $_.Comments.Count "Each thread should have a single comment"}
+    $threadCollection | ForEach-Object {Assert-AreEqual $false $_.Comments.IsDeleted "Each thread should be marked as not deleted"}
+    
+    $threadCollection | ForEach-Object {Assert-AreEqual 8 $_.Properties.Count "Each thread should have 8 properties"}
+    
+    $threadCollection | ForEach-Object {Assert-AreEqual "Source Commit" $_.Properties["CodeReviewSourceCommit"] "Invalid CodeReviewSourceCommit property"}
+    $threadCollection | ForEach-Object {Assert-AreEqual "Target Commit" $_.Properties["CodeReviewTargetCommit"] "Invalid CodeReviewTargetCommit property"}
+    $threadCollection | ForEach-Object { Assert-AreEqual 1 $_.Properties[$startColumnName] "Invalid StartColumn property."}
+    $threadCollection | ForEach-Object { Assert-AreEqual "RightBuffer" $_.Properties[$positionContextName] "Invalid PositionContext property."}
+    $threadCollection | ForEach-Object { Assert-AreEqual "CodeAnalysisIssue" $_.Properties["CodeAnalysisThreadType"] "Invalid CodeAnalysisThreadType property."}
+}
+
+function InitPostCommentsModule 
+{
+    $mockGitClient = New-Object -TypeName "Microsoft.TeamFoundation.SourceControl.WebApi.GitHttpClient"
+    $mockPullRequest = New-Object -TypeName "Microsoft.TeamFoundation.SourceControl.WebApi.GitPullRequest"
+
+    # Legacy PR
+    $mockPullRequest.CodeReviewId = 0
+    $mockPullRequest.LastMergeSourceCommit = "Source Commit"
+    $mockPullRequest.LastMergeTargetCommit = "Target Commit"
+
+    Test-InitPostCommentsModule $mockGitClient $mockPullRequest "artifact uri"
+}
+
+InitPostCommentsModule
+
+# Test 1 (happy path) - Post 2 comments 
+
+# Arrange
+Register-Mock PostDiscussionThreads
```

Comment:
PostDiscussionThreads have very little logic but creates some rather complex objects (Threading Tasks, generics) so I'd rather completely mock it to make the tests simpler to read.